src/api/bitswap/stat.js

```diff
@@ -0,0 +1,22 @@
+'use strict'
+
+const promisify = require('promisify-es6')
+
+module.exports = (arg) => {
+  let send
+
+  if (typeof arg === 'function') {
+    send = arg
+  } else if (typeof arg === 'object') {
+    const requestAPI = require('../../request-api')
+    send = requestAPI(arg)
+  } else {
+    throw new Error('Argument must be a send function or a config object.')
+  }
```

Comment:
Yes, I was thinking in extracting the config processing part in the `IpfsAPI` function, and use it in all modules.